man/systemd-nspawn.xml

```diff
@@ -399,6 +399,18 @@
         is read from it and automatically used, also as formatted hexadecimal characters.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--root-hash-sig=</option></term>
+
+        <listitem><para>Takes a path to a PKCS7 formatted signature of the <option>--root-hash=</option> option.
+        If libcryptsetup (minimum version 2.3.0) and the kernel (minimum version 5.4 with CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG
+        enabled) support it, the dm-verity volume will only be opened if the signature of the root hash hex string is valid
+        and done by a public key present in the kernel keyring. If this option is not specified, but a file with the
+        <filename>.p7</filename> suffix is found next to the image file, bearing otherwise the same name (except if the image
+        has the <filename>.raw</filename> suffix, in which case the signature file must not have it in its name), the
+        signature is read from it and automatically used.</para></listitem>
+      </varlistentry>
+
```

Comment:
The .p7 suffix, is this commonly used for signatures in pkcs7 format? I mean, as I understand pkcs#7 can contain also certificates and what not. Wouldn't a more specific signature be more appropriate? i.e. `.sig` or so?

(I mean, I am not suggesting we should make up our own suffix, we should use the suffix people generally use for this, but I am wondering what that ist, the suffix most people use.)

given that this is the signature of the root hash (and not the signature of the root image itself), I figure the file should by default be derived from the root hash file, so that foo.raw → foo.roothash → foo.roothash.sig is how the name is derived? i.e. foo.roothash.sig sounds more logic to me than foo.sig for this

Reply:
Sure, I can change that - .p7 is just the suffix that we use at the moment, no clue why it was picked.

Again, not an expert, but with some googling I got to the RFC:

https://tools.ietf.org/html/rfc5751#section-3.2.1

I think this description fits:

> application/pkcs7-signature (SignedData)                .p7s

Reply:
sounds good to me